File: src/server/ServerWithCommands.java
package server;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import lib.net.udp.Server;

import java.net.DatagramPacket;
import java.net.SocketException;

public abstract class ServerWithCommands extends Server {
    private final String commandFieldName;

    protected ServerWithCommands(int port, String commandFieldName) throws SocketException {
        super(port);
        objectMapper.findAndRegisterModules();

        this.commandFieldName = commandFieldName;
    }

    protected String getCommandFromJson(DatagramPacket packet) throws JsonProcessingException {
        String result = new String(packet.getData(), 0, packet.getLength());
        JsonNode jsonNodeRoot = objectMapper.readTree(result);
        JsonNode commandNode = jsonNodeRoot.get(commandFieldName);

        return commandNode.asText();
    }
}

File: src/server/ServerWithCommands.kt
package server

import lib.net.udp.JsonHolder
import lib.net.udp.Server
import kotlin.coroutines.CoroutineContext

abstract class ServerWithCommands
protected constructor(port: Int, context: CoroutineContext, private val commandFieldName: String) :
    Server(port, context) {
    protected fun getCommandFromJson(jsonHolder: JsonHolder): String {
        val commandNode = jsonHolder.jsonNodeRoot[commandFieldName]
        return commandNode.asText()
    }
}
